plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/DeltaLakeParquetStatisticsUtils.java

```diff
@@ -71,6 +92,123 @@ public static boolean hasInvalidStatistics(Collection<ColumnChunkMetaData> metad
                         (!metadata.getStatistics().hasNonNullValue() && metadata.getStatistics().getNumNulls() != metadata.getValueCount()));
     }
 
+    @Nullable
+    public static Object toTrinoValue(Type type, @Nullable Object value)
+    {
+        if (value == null) {
+            return null;
+        }
+
+        if (type == SMALLINT || type == TINYINT || type == INTEGER) {
+            return (long) (int) value;
+        }
+        if (type == BIGINT) {
+            return (long) (int) value;
+        }
+        if (type == REAL) {
+            return (long) floatToRawIntBits((float) (double) value);
+        }
+        if (type == DOUBLE) {
+            return value;
+        }
+        if (type instanceof DecimalType) {
+            BigDecimal decimal;
+            checkArgument(value instanceof String || value instanceof Double, "Value must be instance of String or Double");
+            if (value instanceof String) {
+                decimal = new BigDecimal((String) value);
+            }
+            else {
+                decimal = BigDecimal.valueOf((double) value);
+            }
+
+            if (isShortDecimal(type)) {
+                return Decimals.encodeShortScaledValue(decimal, ((DecimalType) type).getScale());
+            }
+            return Decimals.encodeScaledValue(decimal, ((DecimalType) type).getScale());
+        }
+        if (type instanceof VarcharType) {
+            return value;
+        }
+        if (type.equals(DateType.DATE)) {
+            return LocalDate.parse((String) value).toEpochDay();
+        }
+        if (type instanceof TimestampType) {
```

Comment:
I don't think you can write Timestamps in Delta, or TimestampWithTimeZone. You could also have a `verify` that it is a short timestamp